Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp

```diff
@@ -2547,3 +2548,23 @@ eModelLoadState CLuaEngineDefs::EngineStreamingGetModelLoadState(std::uint16_t m
 
     return g_pGame->GetStreaming()->GetStreamingInfo(modelId)->loadState;
 }
+
+void CLuaEngineDefs::EnginePreloadWorldArea(PreloadAreaOption option, CVector position)
+{
+    switch (option)
+    {
+        case PreloadAreaOption::ALL:
+        case PreloadAreaOption::MODELS:
+            g_pGame->GetStreaming()->LoadScene(&position);
+
+            if (option != PreloadAreaOption::ALL)
+            {
+                break;
+            }
+        case PreloadAreaOption::COLLISIONS:
+            g_pGame->GetStreaming()->LoadSceneCollision(&position);
+            break;
+        default:
+            throw std::invalid_argument("Invalid option at argument 1");
```

Comment:
Maybe `if` would be better here?

Reply:
Hmm, I don't see a way to remove the if-statement for the models one. We want to prevent fallthrough if we're not using the ALL option.

Reply:
`switch` is okay here

Reply:
Really shows that I didn't sleep a lot this night lol. Yeah, I'd rather have a switch here than if-statements. Especially if this function may get extended in the future

Reply:
```c++
if (option == PreloadAreaOption::ALL || option == PreloadAreaOption::MODELS)
    g_pGame->GetStreaming()->LoadScene(&position);
    
if (option == PreloadAreaOption::ALL || option == PreloadAreaOption::COLLISIONS)
    g_pGame->GetStreaming()->LoadSceneCollision(&position);
```
`default` is unreachable, argument parser does this check

Reply:
I'll do that, originally I didn't have the enum, just left it after, just in case